### ISSUE TITLE
replace call to functions from the string module in mail

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4995]
|<img alt="nueces's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| nueces reported|
|-|-|
|Trac ID|trac#4995|
|Type|enhancement|
|Created|2011-03-27 17:11:59Z|
|Branch|https://github.com/twisted/twisted/tree/str-methods-mail-4995|

To make the port to python 3k more easy, I'm replacing all the calls to functions from the string module, for their equivalent from the str type object or calling directly to the built-in method of the string variable .

Attachments:

* [mail.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/75d/75dbeede5ade28febc59d2c513c7ca5514e7a736/62c52400efd6294f92cfa8f60c8cd3a119a3cfaf.patch) (1945 bytes) - added by nueces on 2011-03-27 17:12:10Z - patch

<details><summary>Searchable metadata</summary>

```
trac-id__4995 4995
type__enhancement enhancement
reporter__nueces nueces
priority__normal normal
milestone__ 
branch__branches_str_methods_mail_4995 branches/str-methods-mail-4995
branch_author__lvh lvh
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__py3k py3k
time__1301245919000000 1301245919000000
changetime__1302056288000000 1302056288000000
version__None None
owner__lvh lvh

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_2"></a>|@lvh commented|
|-|-|

(In [31451]) Branching to 'str-methods-mail-4995'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_3"></a>|@lvh set owner to @lvh|
|-|-|

Fixed keywords and assigned to me for review

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_5"></a>|@lvh commented|
|-|-|

Buildbot looks clean! http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/str-methods-mail-4995

Some of this stuff really ought to be fixed, but perhaps not in this ticket. One thing that comes to mind is that absurdly long line with the `zfill` in it. Following rationale posted on IRC which nobody sufficiently violently disagreed with to stop me:
 1. It's a genuine, albeit tiny, incremental improvement
 2. It helps the py3k effort

This is going in my merge queue just in case I hear different from some other committer.

Thanks a lot for your patch!

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_6"></a>|@lvh set status to `closed`|
|-|-|

(In [31557]) Replaced calls to functions from the string module in twisted.mail.

Author: nueces
Reviewers: lvh
Fixes: [#4995](https://github.com/twisted/twisted/issues/4995)

twisted.mail was previously using deprecated calls to functions in the string
module, which are deprecated in favor of str methods.
